renderlib/Origins.cpp

```diff
@@ -10,14 +10,18 @@ Origins::update(SceneView& scene)
   if (scene.scene) {
 
     // e.g. find all selected objects in scene and collect up their centers/transforms here.
-    SceneLight& lt = scene.scene->m_lighting.m_sceneLights[1];
+    SceneObject* obj = scene.selection;
 
     // save the initial transform? we could use this to reset things if cancelled.
 
     // this is a copy!!!
-    m_origins = { lt.m_transform };
-    // we want the rotate manipulator to be centered at the target of the light
-    m_origins[0].m_center = lt.m_light->m_Target;
+    m_origins = { obj->m_transform };
+
+    // arealight special case:
+    // we want the rotate manipulator to be centered at the target of the light.
+    // This could be a virtual SceneObject method like getManipulatorCenter()
+    Light& lt = scene.scene->AreaLight();
+    m_origins[0].m_center = lt.m_Target;
```

Comment:
Should `0` be a hardcoded value here?

Reply:
The origins class is "prepared" to hold an array of objects but currently only the first object is ever active.  The idea being that you could have multiple objects selected and then rotate or translate all of them via the Origins class.